src/lib/sync-profile.js

```diff
@@ -0,0 +1,111 @@
+import signalhub from 'signalhubws'
+import swarm from 'webrtc-swarm'
+import * as common from 'masq-common'
+
+import Masq from './masq'
+import { waitForPeer, waitForDataFromPeer, sendEncryptedJSON, decryptJSON, debug } from './utils'
+
+const { dbReady, createPromisifiedHyperDB } = common.utils
+const { importKey } = common.crypto
+
+class SyncProfile {
+  constructor (options) {
+    this.key = null // encryption key
+    this.channel = ''
+    this.hub = null
+    this.sw = null
+    this.peer = null
+    this.db = null
+    this.masq = new Masq()
+    this.hubUrl = options.hubUrl
+    this.swarmOptions = options
+      ? options.swarmOptions || null
+      : null
+  }
+
+  async joinSecureChannel (channel, rawKey) {
+    debug('joinSecureChannel', channel)
+    this.key = await importKey(Buffer.from(rawKey, 'base64'))
+
+    this.hub = signalhub(channel, this.hubUrl)
+    this.hub.on('error', () => {})
+
+    this.sw = swarm(this.hub, this.swarmOptions)
+    this.sw.on('disconnect', () => { })
+
+    const peer = await waitForPeer(this.sw)
+    this.peer = peer
+  }
+
+  async pullProfile () {
+    let data
+
+    await sendEncryptedJSON({ msg: 'pullProfile' }, this.key, this.peer)
+
+    data = await waitForDataFromPeer(this.peer)
+    const { msg, id, key, publicProfile } = await decryptJSON(data, this.key)
+    debug('pullProfile received', msg, id, key)
```

Comment:
isn't it 'pullProfile' that is received ?

Reply:
I meant "pullProfile received" like "pullProfile (the function's name) received:"